tests/v2/validation/airgap/k3s_custom_cluster_test.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/pipeline/rancherha/corralha"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/rancher/rancher/tests/v2/validation/registries"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type AirGapK3SCustomClusterTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	session            *session.Session
+	kubernetesVersions []string
+	cnis               []string
+	corralImage        string
+	corralAutoCleanup  bool
+	registryFQDN       string
+	advancedOptions    provisioning.AdvancedOptions
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TearDownSuite() {
+	a.session.Cleanup()
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	a.session = testSession
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	corralRancherHA := new(corralha.CorralRancherHA)
+	config.LoadConfig(corralha.CorralRancherHAConfigConfigurationFileKey, corralRancherHA)
+
+	registriesConfig := new(registries.Registries)
+	config.LoadConfig(registries.RegistriesConfigKey, registriesConfig)
+
+	a.kubernetesVersions = clustersConfig.K3SKubernetesVersions
+	a.cnis = clustersConfig.CNIs
+	a.advancedOptions = clustersConfig.AdvancedOptions
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+	listOfCorrals, err := corral.ListCorral()
+	require.NoError(a.T(), err)
+
+	corralConfig := corral.CorralConfigurations()
+
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	corralPackage := corral.CorralPackagesConfig()
+	a.corralImage = corralPackage.CorralPackageImages[corralPackageAirgapCustomClusterName]
+	a.corralAutoCleanup = corralPackage.Cleanup
+
+	_, corralExist := listOfCorrals[corralRancherHA.Name]
+	if corralExist {
+		bastionIP, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryIP)
+		require.NoError(a.T(), err)
+
+		err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+		require.NoError(a.T(), err)
+
+		registryFQDN, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryFQDN)
+		require.NoError(a.T(), err)
+		logrus.Infof("registry fqdn is %s", registryFQDN)
+		a.registryFQDN = registryFQDN
+	} else {
+		a.registryFQDN = registriesConfig.ExistingNoAuthRegistryURL
+	}
+
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TestProvisioningK3SCustomCluster() {
+	nodeRoles := map[int]string{
+		1: "--etcd --controlplane --worker",
+	}
+
+	var name string
+	for _, kubeVersion := range a.kubernetesVersions {
+		name = "Airgap Provision K3S Custom Cluser Kubernetes version: " + kubeVersion
```

Comment:
For regex purposes, do we want to perhaps shorten the name down? I don't have a strong preference on it, just seems like a lot to type out, but again the provisioning test names are long as well...

Reply:
Sure I get that, what would you suggest? Airgap K3S Custom Cluster K8s version: ...

Reply:
Sounds good I'll make the change

Reply:
Made the update